### PR TITLE
Use `kube play` and `play kube` instead of containers format

### PR DESCRIPTION
**Pull request by richm**
Use `kube play` and `play kube` with the Kubernetes YAML format
as the way to configure pods/containers on the managed node.

### CODE REVIEW

**Comment by richm** [comment]
@maxamillion FYI
NOTE: This relies on the latest podman 4.2 which is not GA yet, and as such relies on the copr repo at `rhcontainerbot/podman-next`.  I expect integration test failures on platforms which are not supported by the copr repo.
Once podman 4.2 is GA on our supported platforms I will drop the copr repo.

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
oh, and the other tests fail due to the fact that CI doesn't yet support using collections from roles

File: README.md [hunk 1]
```diff
@@ -16,44 +16,65 @@ ansible-galaxy collection install -vv -r meta/collection-requirements.yml
...
+```yaml
+    podman_kube_specs:
+      - state: started
+        kube_file_content:
+          apiVersion: v1
+          kind: Pod
+          metadata:
+            name: myappname
+```
+This will be copied to the file `/etc/containers/kubernetes.d/myappname.yml` on
```
Comment (vrothberg) [inline README.md:64]: Could we rename it to `ansible-kubernetes.d`?  Just to make sure, future Podman maintainers don't create a conflict with the rather generic `kubernetes.d`.

**Review by vrothberg — COMMENTED** [review]
Other than the naming nit, LGTM

Impressive work, @richm!

File: README.md (hunk 1 above)
Comment (richm) [reply README.md:64]: done

**Comment by Tronde** [comment]
> @maxamillion FYI NOTE: This relies on the latest podman 4.2 which is not GA yet, and as such relies on the copr repo at `rhcontainerbot/podman-next`. I expect integration test failures on platforms which are not supported by the copr repo. Once podman 4.2 is GA on our supported platforms I will drop the copr repo.

Did I get it right, that with this PR merged I would need podman 4.2 on the managed node to be able to use this role? IMHO the `README.md` should mention podman 4.2 as a requirement then.

I use podman 3.0.1 on Debian bullseye, 4.0.2 on RHEL 8.6 and version 4.1.1 on fedora 36. My guess is that I wouldn't be able to use this role on the mentioned podman versions, right?

**Comment by richm** [comment]
> > @maxamillion FYI NOTE: This relies on the latest podman 4.2 which is not GA yet, and as such relies on the copr repo at `rhcontainerbot/podman-next`. I expect integration test failures on platforms which are not supported by the copr repo. Once podman 4.2 is GA on our supported platforms I will drop the copr repo.
> 
> Did I get it right, that with this PR merged I would need podman 4.2 on the managed node to be able to use this role? IMHO the `README.md` should mention podman 4.2 as a requirement then.

Done

> 
> I use podman 3.0.1 on Debian bullseye, 4.0.2 on RHEL 8.6 and version 4.1.1 on fedora 36. My guess is that I wouldn't be able to use this role on the mentioned podman versions, right?

Right.  Note that podman 4.2 should be GA soon and should be going into rhel 8 and 9 in the near future.

Why did we do this?  Why did we design a podman role that would ignore 90% of podman users?  It was felt that the Kubernetes YAML api was the preferred future direction, and we didn't want to support two different APIs indefinitely.

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
[citest]

**Review by spetrosi — APPROVED** [review]
lgtm

File: README.md [hunk 2]
```diff
@@ -16,44 +18,65 @@ ansible-galaxy collection install -vv -r meta/collection-requirements.yml
...
+* `kube_file_content` - This is either a string in Kubernetes YAML format, or
+  a `dict` in Kubernetes YAML format.  It will be used as the contents of
+  `kube_file` on the managed node.
+* `kube_file` - If you specify either `kube_file_src` or `kube_file_content`, you
+  do not have to specify this.  It is highly recommended to omit `kube_file` and
+  instead specify either `kube_file_src` or `kube_file_content` and let the role
+  manage the file path and name.
+  * The file basename will be the `metadata.name` value from the K8s yaml, with a
+    `.yml` suffix appended to it.
+  * The directory will be `/etc/containers/ansible-kubernetes.d` for system services
```
Comment (nhosoi) [inline README.md:53]: A very, very minor thing... A period at the end of the line is missing. :)

I'd think one of `kube_file_src`, `kube_file_content` and `kube_file` is supposed to be specified in `podman_kube_specs`. May I ask what happens when 2 or more are given or one of them is set? Does it issue an error or silently processed somehow?

Comment (richm) [reply README.md]: > A very, very minor thing... A period at the end of the line is missing. :)

Fixed

> I'd think one of `kube_file_src`, `kube_file_content` and `kube_file` is supposed to be specified in `podman_kube_specs`. May I ask what happens when 2 or more are given or one of them is set? Does it issue an error or silently processed somehow?

clarified in the docs

**Review by nhosoi — APPROVED** [review]
lgtm

**Pull request merged**